### ISSUE TITLE
Input object parsing with number after underscore

### ISSUE THREAD

**Issue by RealSilo** [rmosolgo/graphql-ruby#1839]
I have an `address_1` field in my Postgres DB schema for my address table. My GraphQL mutation has `address1` field and the input type definition on the BE has `argument :address_1, String, 'Address', required: true`.

No error is raised but in the resolve method when I access the param I have it as `address1` which is of course not ideal. Is this the intended behavior? If so, why?

**Comment by RealSilo** [comment]
```
module Types
  class AddressType < GraphQL::Schema::Object
    field :address_1, String, 'Address 1', null: false
    ...
  end
end
```

Customer model includes `accepts_nested_attributes_for :addresses`

```
module Mutations
  class UpdateCustomer < Mutations::BaseMutation
    argument :id, ID, required: true
    ......
    argument :address, Types::AddressInput, required: false

    field :customer, Types::CustomerType, null: true
    field :errors, [Types::ErrorType], null: false

    def resolve(**params)
      # byebug inserted here and photo added below
      address_params = params.delete(:address).to_h
      address_params[:address_1] = address_params.delete(:address1)
      params[:addresses_attributes] = [address_params]

      customer = Customer.find(params[:id])

      authorize(context[:current_user], customer)

      if customer.update_attributes(params)
        {
          customer: customer,
          errors: []
        }
      else
        {
          customer: nil,
          errors: build_errors(customer.errors)
        }
      end
    end

    private

    def authorize(current_user, customer)
      return if CustomerPolicy.new(current_user, customer).update?
      raise GraphQL::ExecutionError, :forbidden
    end
  end
end
```

<img width="1052" alt="screen shot 2018-09-08 at 11 44 22" src="https://user-images.githubusercontent.com/10554687/45252767-cb103400-b35c-11e8-8d9c-53b9f4b34683.png">

<img width="498" alt="screen shot 2018-09-08 at 11 47 11" src="https://user-images.githubusercontent.com/10554687/45252778-f561f180-b35c-11e8-905d-0d290a4bc847.png">


Also pls let me know if there is an easier way to do nested mutations. Thx!

**Comment by cbzehner** [comment]
tl;dr: I had the same issue and was able to resolve it by using the `as:` syntax to provide an alias inside the `resolve` function to provide `address_1` as an alias. See the example below.


```diff
- argument :address_1, String, required: false
- argument :address_2, String, required: false
- argument :address_3, String, required: false
+ argument :address_1, String, required: false, as: :address_1
+ argument :address_2, String, required: false, as: :address_2
+ argument :address_3, String, required: false, as: :address_3
```

This is because Ruby's `underscore()` method for the `String` class doesn't treat names like `address1` as two tokens -- I haven't looked into it but it's potentially a bug since `camelize()` _does_ combine the two. Bears further investigation since I'd expect the below functions to be the reverse of each other.

```ruby
> 'address_1'.camelize(:lower)
=> "address1"
> 'address1'.underscore()
=> "address1"
```

**Comment by rmosolgo** [comment]
>  I'd expect the below functions to be the reverse of each other

Me too, that seems like the source of this issue 😖

**Comment by lsanwick** [comment]
Just got stung by this as well, the fix is simple, but the problem is definitely not intuitive. :(

**Comment by rmosolgo** [comment]
`as: ...` won't be necessary in 1.10.4, after #2792 lands. Instead GraphQL-Ruby will use the symbol _as given_.
